cabal.project

```diff
@@ -44,43 +61,25 @@ benchmarks: true
 -- The only sensible test display option
 test-show-details: streaming
 
-source-repository-package
-  type: git
-  location: https://github.com/input-output-hk/cardano-base
-  tag: 7e3ddba98a61900181fe63cdd4c9ed9708d1a6a7
-  --sha256: 1ci0aa8zkcld0d4i6xldvffhppf8bg0ys8y2jys912r7i5sxq5pa
-  subdir:
-    base-deriving-via
-    binary
-    binary/test
-    cardano-crypto-class
-    cardano-crypto-praos
-    heapwords
-    measures
-    slotting
-    strict-containers
+allow-newer:
+  monoidal-containers:aeson,
+  size-based:template-haskell,
+  Unique:hashable
 
-source-repository-package
-  type: git
-  location: https://github.com/input-output-hk/cardano-crypto
-  tag: f73079303f663e028288f9f4a9e08bcca39a923e
-  --sha256: 1n87i15x54s0cjkh3nsxs4r1x016cdw1fypwmr68936n3xxsjn6q
+constraints:
+  -- bizarre issue: in earlier versions they define their own 'GEq', in newer
+  -- ones they reuse the one from 'some', but there isn't e.g. a proper version
+  -- constraint from dependent-sum-template (which is the library we actually use).
+  , dependent-sum > 0.6.2.0
 
-source-repository-package
-  type: git
-  location: https://github.com/input-output-hk/cardano-prelude
-  tag: 533aec85c1ca05c7d171da44b89341fb736ecfe5
-  --sha256: 0z2y3wzppc12bpn9bl48776ms3nszw8j58xfsdxf97nzjgrmd62g
-  subdir:
-    cardano-prelude
-    cardano-prelude-test
+  -- Plutus dependency
+  , algebraic-graphs >= 0.7
 
-source-repository-package
-  type: git
-  location: https://github.com/input-output-hk/goblins
-  tag: cde90a2b27f79187ca8310b6549331e59595e7ba
-  --sha256: 17c88rbva3iw82yg9srlxjv2ia5wjb9cyqw44hik565f5v9svnyg
+  , protolude >= 0.3.2
```

Comment:
these constraints are here rather than in the cabal file? easier transition?

Reply:
They're for packages not in this repository...